root/components/common-macros.tt

```diff
@@ -229,27 +229,24 @@ END -%]
 
 [%~ MACRO entity_exists(entity) BLOCK; entity.gid.defined; END -%]
 
-[%~ MACRO link_entity(entity, action, text, credited_as, no_escape) BLOCK;
-    # no_escape is only available for artists and recordings
+[%~ MACRO link_entity(entity, action, text, opts) BLOCK;
+    # opts is an optional hash reference; possible keys:
+    # - show_disambiguation (omit for default behaviour)
+    # - already_html: the text argument is HTML; no entity encoding
+    #   will be done.
     show_disambiguation = text == '';
-    IF (NOT text.defined OR text == '');
-        text = credited_as;
-    END;
+    SET show_disambiguation = opts.show_disambiguation
+      IF opts.show_disambiguation.exists;
     IF entity.gid;
-      IF    entity.isa('MusicBrainz::Server::Entity::Artist'); link_artist(entity, action, text, no_escape);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Area'); link_area(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Collection'); link_collection(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Work'); link_work(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Event'); link_event(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Instrument'); link_instrument(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Label'); link_label(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Place'); link_place(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Release'); link_release(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::ReleaseGroup'); link_release_group(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Recording'); link_recording(entity, action, text, no_escape);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::Series'); link_series(entity, action, text);
-      ELSIF entity.isa('MusicBrainz::Server::Entity::URL'); link_url(entity, action, text);
-      END;
+      ent_type = get_entity_type(entity);
+      INCLUDE _link_mbid_entity entity=entity
+          type=ent_type action=action content=text
+          noescape=opts.already_html
+          hover=ent_type == 'artist' ?
+              artist.sort_name _ bracketed(artist.comment) : ''
+          default_content=(ent_type == 'area' OR ent_type == 'instrument') ?
+              entity.l_name : ''
+          namevar=(ent_type == 'artist' OR ent_type == 'recording');
```

Comment:
I’m wondering (but that would be for a different PR) if we shouldn’t extend name-variation to all entity types (except URLs, probably, which don’t actually have a name). Perhaps @jesus2099 has an opinion about that?